Vector.c

```diff
@@ -200,15 +192,16 @@ void Vector_insertionSort(Vector* this) {
    assert(Vector_isConsistent(this));
 }
 
-static void Vector_checkArraySize(Vector* this) {
+static void Vector_resizeIfNecessary(Vector* this, int newSize) {
+   assert(newSize >= 0);
    assert(Vector_isConsistent(this));
-   if (this->items >= this->arraySize) {
-      //int i;
-      //i = this->arraySize;
-      this->arraySize = this->items + this->growthRate;
+   if (newSize > this->arraySize) {
+      int oldSize = this->arraySize;
+      this->arraySize = newSize + this->growthRate;
       this->array = (Object**) xRealloc(this->array, sizeof(Object*) * this->arraySize);
-      //for (; i < this->arraySize; i++)
-      //   this->array[i] = NULL;
+      for (int i = oldSize; i < this->arraySize; i++) {
```

Comment:
I think we can use `xReallocArrayZero(this->array, oldSize, this->arraySize, sizeof(Object*))` here